src/utils/upload/processUpload.js

```diff
@@ -6,61 +6,14 @@ import {
 } from 'redux/actions/samples';
 
 import UploadStatus from 'utils/upload/UploadStatus';
-import loadAndCompressIfNecessary from 'utils/upload/loadAndCompressIfNecessary';
 import { inspectFile, Verdict } from 'utils/upload/fileInspector';
 import fetchAPI from 'utils/http/fetchAPI';
 
 import { sampleTech } from 'utils/constants';
 import fileUploadUtils from 'utils/upload/fileUploadUtils';
-import processMultipartUpload from 'utils/upload/processMultipartUpload';
 import endUserMessages from 'utils/endUserMessages';
 import pushNotificationMessage from 'utils/pushNotificationMessage';
-
-const prepareAndUploadFileToS3 = async (
-  file, uploadUrlParams, type, abortController, onStatusUpdate = () => { },
-) => {
-  let parts = null;
-  const { signedUrls, uploadId, fileId } = uploadUrlParams;
-
-  const uploadedPartSizes = new Array(signedUrls.length).fill(0);
-  const totalSize = file.size;
-
-  const createOnUploadProgressForPart = (partIndex) => (progress) => {
-    uploadedPartSizes[partIndex] = progress.loaded;
-    const totalUploaded = _.sum(uploadedPartSizes);
-    const percentProgress = Math.floor((totalUploaded * 100) / totalSize);
-
-    onStatusUpdate(UploadStatus.UPLOADING, percentProgress);
-  };
```

Comment:
The previous upload progress tracking system was way more detailed than what I replaced it with. 

What it did was:
- For each file: It tracked the progress on each individual part's upload, then gave the file's progress by joining all of this file's individual part's progresses.

I replaced it with a way dumber system: 
- It doesn't track progress within each part. It only tracks which parts finished. This means that for files smaller than the minimum chunk size (under 128MB in what I set, we can change that), the progress bar won't show up.

This is because after my changes we can no longer apply the more advanced tracking easily:
- Before, we would load a file on memory, compress it and then track the progress on the upload size of the compressed file. Since we have the whole compressed file before beginning the upload, we will know how far we are from finishing the upload during the whole way.
- Now, the compression takes place in a stream. We don't really know how far we are from finishing the upload in terms of file size until the whole file is compressed (unless we could somehow guess what the compressed file's size will be, but I don't think going into this is worth it). What we do know is how many parts the file is split into, so that's why I am tracking the file parts.

Reply:
If we think people will really feel the loss of a progress bar for files < 128MB (or whatever size we choose), we can add an edge case for these files (if parts to upload is 1, then use the progress of the axios upload)